scripts/events/onEditButtonClick.js

```diff
@@ -0,0 +1,53 @@
+import { CLICK_EVENT } from "/scripts/constants.js";
+import toggleButton from "/scripts/factory/toggleButton.js";
+import saveEvent from "/scripts/events/saveEvent.js";
+import onTodoDone from "/scripts/events/onTodoDone.js";
+import revertEvent from "/scripts/events/revertEvent.js";
+import disableEditing from "/scripts/factory/disableEditing.js";
+
+function onEditButtonClick(
+  buttons,
+  inputElement,
+  toolbar,
+  tasks,
+  id,
+  editEventHandler
+) {
+  const prevContent = inputElement.innerText;
+  disableEditing(inputElement);
+  inputElement.focus();
+  toggleButton(...buttons);
+
+  function saveEventHandler() {
+    saveEvent(buttons, inputElement, tasks, id, eventHandlers);
+  }
+
+  function saveAndDoneEventHandler() {
+    onTodoDone(
+      buttons,
+      inputElement,
+      toolbar,
+      tasks,
+      id,
+      editEventHandler,
+      eventHandlers
```

Comment:
why does the `onTodoDone` function takes multiple event handlers?

Reply:
@sam-wtag Because we need to remove all the events after onTodoDone at onTodoDone.js line-42.
`removeEventListeners(buttons, ...eventHandlers);`